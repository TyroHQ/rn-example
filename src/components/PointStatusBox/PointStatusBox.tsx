import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "../Placeholder";

const PointStatusBoxContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class PointStatusBox extends React.Component<{}> {
  static defaultProps: PointStatusBox["props"];
  render() {
    return (
      <Placeholder>
        Point Status
        <Placeholder>Input placeholder</Placeholder>
        <Placeholder>Header text</Placeholder>
      </Placeholder>
    );
  }
}
