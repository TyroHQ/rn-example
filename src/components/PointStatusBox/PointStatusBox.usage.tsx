import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { PointStatusBox } from "./";

export const basic = () => <PointStatusBox />;
