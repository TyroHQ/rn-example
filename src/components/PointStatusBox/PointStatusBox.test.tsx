import { shallow } from "enzyme";
import * as usage from "./PointStatusBox.usage";

describe("<PointStatusBox />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
