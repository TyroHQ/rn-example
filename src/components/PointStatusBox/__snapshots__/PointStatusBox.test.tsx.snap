// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PointStatusBox /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  Point Status
  <Placeholder
    orientation="row"
  >
    Input placeholder
  </Placeholder>
  <Placeholder
    orientation="row"
  >
    Header text
  </Placeholder>
</Placeholder>
`;
